### PR TITLE
Add contributing guide & remove tracing on tags

### LINKED ISSUE #154: Add guide for develop installations
We need to add guidances for developers to install from github and prepare for development, both in README and docs.

### PR DESCRIPTION
**Pull request by wjsi**
## What do these changes do?

Add contributing guide for potential contributors. Also removes profiling envs when there is a tag, diffs can be seen between [exec log before change](https://travis-ci.org/mars-project/mars/jobs/483830569) and [exec log after change](https://travis-ci.org/wjsi/mars/jobs/483958251), in the ``build_ext`` step.

## Related issue number

Fixes #154

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+.. code-block:: bash
+
+    python -c "import mars; print(mars.__version__)"
+
+If this command correctly outputs your Mars version, Mars is correctly
+installed.
+
+Rebuilding Cython Code
+``````````````````````
+Mars uses Cython to accelerate part of its code. After you changes Cython
```
Comment (chineking) [inline docs/source/contributing.rst]: changes -> change

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+files, or ``*.pyx``, you need to compile them into binaries by executing the
```
Comment (chineking) [inline docs/source/contributing.rst]: Why is or? maybe In my opinion, including ``*.pxd`` or ``*.pyx``  is better?

File: docs/source/index.rst [hunk 3]
```diff
@@ -63,3 +63,9 @@ it's fairly simple to migrate from a single machine to a cluster due to the incr
    distributed/schedule-policy
    distributed/states
    distributed/worker-schedule
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Contributor Documentation
```
Comment (chineking) [inline docs/source/index.rst]: Contribution Guide seems a better name.

File: .travis.yml [hunk 4]
```diff
@@ -43,14 +43,15 @@ before_install:
 # command to install dependencies
 install:
   - export DEFAULT_VENV=$VIRTUAL_ENV
+  - if [ ! -z "$TRAVIS_TAG" ]; then unset CYTHON_TRACE; fi
```
[the remark this answers is not in the document]
Comment (wjsi) [reply .travis.yml]: See ``gcc`` commands in the urls above, both triggered by tagging. I don't think we need to actually upload to test.pypi.org.

[the remark this answers is not in the document]
Comment (chineking) [reply .travis.yml]: OK, since tag is only created when new version released, I just want to confirm that this command is workable.

**Comment by wjsi** [comment]
No back port needed, as there are no line-tracing in v0.1.

**Pull request merged**